rclpy/test/test_executor.py

```diff
@@ -402,6 +418,72 @@ def timer_callback():
 
         timer.cancel()
 
+    def test_executor_spin_until_complete_condition_done(self):
+        self.assertIsNotNone(self.node.handle)
+        executor = SingleThreadedExecutor(context=self.context)
+        executor.add_node(self.node)
+
+        def timer_callback():
+            pass
+        timer = self.node.create_timer(0.1, timer_callback)
+
+        condition_var = False
+
+        def set_condition():
+            nonlocal condition_var
+            condition_var = True
+
+        def condition():
+            nonlocal condition_var
+            return condition_var
+
+        # Condition complete timeout_sec > 0
+        self.assertFalse(condition())
+        t = threading.Thread(target=lambda: set_condition())
+        t.start()
+        executor.spin_until_complete(condition, timeout_sec=1.0)
+        self.assertTrue(condition())
```

Comment:
This is a fairly weak test in my opinion, as you could comment out line 444 and have it pass most of the time. Also, if `spin_until_complete()` was broken (and did not return before the timeout if the condition was true) this would still pass.